src/main/java/seedu/address/logic/commands/MarkPersonDescriptor.java

```diff
@@ -159,4 +159,24 @@ public void setTags(Set<Tag> tags) {
     public Optional<Set<Tag>> getTags() {
         return (tags != null) ? Optional.of(Collections.unmodifiableSet(tags)) : Optional.empty();
     }
+
+    @Override
+    public boolean equals(Object other) {
+        // for same object
+        if (other == this) {
+            return true;
+        }
+
+        // checks for null object
+        if (!(other instanceof MarkPersonDescriptor)) {
+            return false;
+        }
+
+        // check for same values in fields
+        MarkPersonDescriptor m = (MarkPersonDescriptor) other;
+        return getHomework().equals(m.getHomework())
+                && getHomeworkIndex().equals(m.getHomeworkIndex())
+                && getAttendance().equals(m.getAttendance())
+                && getAttendanceIndex().equals(m.getAttendanceIndex());
+    }
```

Comment:
wouldn't it be ok for MarkPersonDescriptor to only keep HomeworkIndex and AttendanceIndex? doesn't seem like it needs to know about the description of the items?

Reply:
Did the necessary changes